subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -953,23 +915,77 @@ static void pe_chunk_received_run(void *obj)
 	}
 }
 
-/**
- * @brief Suspend State
+/*
+ * @brief Super State for any message that requires
+ *	  Sender Response Timer functionality
  */
-static void pe_suspend_entry(void *obj)
+static void pe_sender_response_run(void *obj)
 {
-	LOG_INF("PE_SUSPEND");
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	enum usbc_pe_state current_state = pe_get_state(dev);
+
+	/* Start the Sender Response Timer after the message is sent */
+	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		/* Start Sender Response Timer */
+		usbc_timer_start(&pe->pd_t_sender_response);
+	}
+
+	/* Check if the Sender Response Timer has expired */
+	if (usbc_timer_expired(&pe->pd_t_sender_response)) {
+		/* Inform the DPM of the sender response timeout */
+		policy_notify(dev, SENDER_RESPONSE_TIMEOUT);
```

Comment:
Is there a part of the spec that requires we notify the DPM of sender response timeouts?  

I suppose it could help clear up confusion for states where the spec doesn't really have anything to do after the timeout (ex. when getting source caps).  But it could also mean we sent notifications for things the DPM didn't direct us to do (ex. soft resets)

Reply:
You are correct, the spec does not require the DPM be notified of a sender response timeout. I'm open to keeping or removing the notification.

Reply:
I don't know that the DPM would know what to do with the notification... it's pretty dependent on the message being sent.  I think it might be for the best to leave it out?

Reply:
I agree that any non-DPM request failures, such as soft reset, should be handled transparently by the stack. I've removed the SENDER_RESPONE_TIMEOUT policy notification.